tests/test_equationbc.py

```diff
@@ -0,0 +1,120 @@
+import pytest
+from firedrake import *
+from math import isclose
+from irksome import GaussLegendre, RadauIIA, Dt, MeshConstant, TimeStepper
+
+
+@pytest.mark.parametrize("stage_type", ["deriv"])
+@pytest.mark.parametrize("butcher_tableau", [GaussLegendre(3)])
+def test_1d_heat_equationbc(butcher_tableau, stage_type):
+
+    # Boundary values
+    u_0 = Constant(2.0)
+    u_1 = Constant(3.0)
+
+    N = 100
+    x0 = 0.0
+    x1 = 10.0
+    msh = IntervalMesh(N, x1)
+    V = FunctionSpace(msh, "CG", 1)
+    MC = MeshConstant(msh)
+    dt = MC.Constant(10.0 / N)
+    t = MC.Constant(0.0)
+    (x,) = SpatialCoordinate(msh)
+
+    # Method of manufactured solutions copied from Heat equation demo.
+    S = Constant(2.0)
+    C = Constant(1000.0)
+    B = (x - Constant(x0)) * (x - Constant(x1)) / C
+    R = (x * x) ** 0.5
+    # Note end linear contribution
+    uexact = (
+        B * atan(t) * (pi / 2.0 - atan(S * (R - t)))
+        + u_0
+        + ((x - x0) / x1) * (u_1 - u_0)
+    )
+    rhs = Dt(uexact) - div(grad(uexact))
+    u = Function(V)
+    u.interpolate(uexact)
+
+    v = TestFunction(V)
+    F = (
+        inner(Dt(u), v) * dx
+        + inner(grad(u), grad(v)) * dx
+        - inner(rhs, v) * dx
+    )
+    bc = [
```

Comment:
Here or elsewhere, we should come up with a test that makes sure `EquationBC` works correctly if we have time-dependent data -- some kind of explicit `t` dependence.

Reply:
`bdrydata` depends on `uexact` and `uexact` is an expression involving `t`.